kalite/distributed/static/js/distributed/rating/views.js

```diff
@@ -0,0 +1,238 @@
+var Backbone = require("../base/backbone");
+var $ = require("../base/jQuery");
+var _ = require("underscore");
+var BaseView = require("../base/baseview");
+var Handlebars = require("../base/handlebars");
+
+/*
+    Container view for feedback form.
+    See the design document here: https://docs.google.com/a/learningequality.org/document/d/1W_2vv1cZU5wZp_MLjvPy6jHwkRVPW_iTJhHEpSOYrCs/edit?usp=drive_web
+*/
+module.exports = BaseView.extend({
+
+    template: require("./hbtemplates/rating.handlebars"),
+
+    events: {
+        "click .rating-submit": "submit_rating",
+        "click .rating-edit": "edit_callback",
+        "click .rating-delete": "delete_callback"
+    },
+
+    initialize: function(options) {
+        /*
+            Prepare self and subviews.
+        */
+        this.model = options.model || function() {
+            var model = new Backbone.Model();
+            model.url = "/api/not/gonna/work";
+            return model;
+        }();
+        _.bindAll(this, "delete_callback", "submit_rating", "edit_callback", "renderQuality", "renderDifficulty", "renderTextArea", "renderAll", "render");
+    },
+
+    render: function() {
+        if( this.model.isNew() || this.model.get("quality") === 0) {
+            this.renderQuality();
+        } else {
+            this.renderAll();
+        }
+    },
+
+    add_subview_and_once: function(view_class, name, options, event, callback) {
+        /*
+            Repeated logic for displaying each subwidget once, then removing it after user has interacted with it
+            and displaying the next widget in the sequence.
+        */
+        this[name] = this.add_subview(view_class, options);
+        var self = this;
+        this.model.once(event, function() {
+            self[name].remove();
+            callback();
+        });
+    },
+
+    renderQuality: function() {
+        /*
+            Render the first star rating, then wait until it's been rated to show the next step.
+            Called when the view's model is not fetched successfully -- particularly when it doesn't yet exist.
+        */
+        this.$el.html(this.template());
+        this.$(".rating-submit").hide();
+        this.$(".rating-edit").hide();
+        this.$(".rating-skip").hide();
+        this.$(".rating-delete").hide();
+        this.add_subview_and_once(StarView, "star_view_quality",
+            {el: this.$("#star-container-quality"), model: this.model, rating_attr: "quality"},
+            "change:quality", this.renderDifficulty);

```

Comment:
Rather than always explictly calling `renderDifficulty`, just have it raise its own event and have another function attach the next function.